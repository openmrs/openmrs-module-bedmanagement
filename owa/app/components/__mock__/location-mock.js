Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});