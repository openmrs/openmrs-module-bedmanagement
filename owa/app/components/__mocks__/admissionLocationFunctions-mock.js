import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
