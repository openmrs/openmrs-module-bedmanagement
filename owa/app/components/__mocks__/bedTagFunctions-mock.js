import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
