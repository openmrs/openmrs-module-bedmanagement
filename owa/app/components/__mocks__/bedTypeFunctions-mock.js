import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
