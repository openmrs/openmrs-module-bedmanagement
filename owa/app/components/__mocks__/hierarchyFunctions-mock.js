import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;