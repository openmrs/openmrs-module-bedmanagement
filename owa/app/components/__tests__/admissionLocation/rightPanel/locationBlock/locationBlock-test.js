import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import LocationBlock from 'components/admissionLocation/rightPanel/locationBlock';
import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';

require('components/__mocks__/location-mock');
const testProps = {
    admissionLocationFunctions: admissionLocationFunctionsMock,
    admissionLocations: admissionLocationFunctionsMock.getAdmissionLocations()
};

describe('LocationBlock', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {};

        mock.onDelete('https://192.168.33.10/openmrs/ws/rest/v1/admissionLocation/baf7bd38-d225-11e4-9c67-080027b662ec')
            .reply(200, data);

    });

    it('Should display location block properly', () => {
        const generalWardLocationBlock = shallow(<LocationBlock activeUuid="baf7bd38-d225-11e4-9c67-080027b662ec"
            admissionLocation={testProps.admissionLocations['baf7bd38-d225-11e4-9c67-080027b662ec']}
            admissionLocationFunctions={testProps.admissionLocationFunctions} />);

        expect(generalWardLocationBlock.find('span').text().trim()).toBe('General Ward');
        expect(shallowToJson(generalWardLocationBlock)).toMatchSnapshot();

        const labourWardLocationBlock = shallow(<LocationBlock activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
            admissionLocation={testProps.admissionLocations['bb0e512e-d225-11e4-9c67-080027b662ec']}
            admissionLocationFunctions={testProps.admissionLocationFunctions} />);

        expect(labourWardLocationBlock.find('span').text().trim()).toBe('Labour Ward');
        expect(shallowToJson(labourWardLocationBlock)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnClickHandler = jest.spyOn(LocationBlock.prototype, 'onClickHandler');
        const spyOnEditWardClickHandler = jest.spyOn(LocationBlock.prototype, 'editWardClickHandler');
        const spyOnDeleteHandler = jest.spyOn(LocationBlock.prototype, 'onDeleteHandler');
        const spySetState = jest.spyOn(testProps.admissionLocationFunctions, 'setState');
        const generalWardLocationBlock = mount(<LocationBlock activeUuid="baf7bd38-d225-11e4-9c67-080027b662ec"
            admissionLocation={testProps.admissionLocations['baf7bd38-d225-11e4-9c67-080027b662ec']}
            admissionLocationFunctions={testProps.admissionLocationFunctions} />);

        generalWardLocationBlock.find('.location.block').simulate('click');
        expect(spyOnClickHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'listing',
            pageData: {},
            activeUuid: 'baf7bd38-d225-11e4-9c67-080027b662ec'
        });

        generalWardLocationBlock.find('.fa-pencil').simulate('click');
        expect(spyOnEditWardClickHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'addEditLocation',
            pageData: {
                operation: 'edit'
            },
            activeUuid: 'baf7bd38-d225-11e4-9c67-080027b662ec'
        });

        const labourWardLocationBlock = mount(<LocationBlock activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
            admissionLocation={testProps.admissionLocations['bb0e512e-d225-11e4-9c67-080027b662ec']}
            admissionLocationFunctions={testProps.admissionLocationFunctions} />);
        labourWardLocationBlock.find('.fa-trash').simulate('click');
        expect(spyOnDeleteHandler).toHaveBeenCalled();
    });

});