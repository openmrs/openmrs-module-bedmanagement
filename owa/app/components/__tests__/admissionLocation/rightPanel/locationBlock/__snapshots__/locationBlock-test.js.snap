// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationBlock Should display location block properly 1`] = `
<div
  className="location block"
  onClick={[Function]}
>
  <div
    className="left-block"
  >
    <span>
      General Ward
    </span>
  </div>
  <ul
    className="right-block"
  >
    <li>
      <a
        href="javascript:void(0);"
        onClick={[Function]}
        title="edit"
      >
        <i
          aria-hidden="true"
          className="fa fa-pencil"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`LocationBlock Should display location block properly 2`] = `
<div
  className="location block"
  onClick={[Function]}
>
  <div
    className="left-block"
  >
    <span>
      Labour Ward
    </span>
  </div>
  <ul
    className="right-block"
  >
    <li>
      <a
        href="javascript:void(0);"
        onClick={[Function]}
        title="edit"
      >
        <i
          aria-hidden="true"
          className="fa fa-pencil"
        />
      </a>
    </li>
    <li>
      <a
        href="javascript:void(0);"
        onClick={[Function]}
        title="delete"
      >
        <i
          aria-hidden="true"
          className="fa fa-trash"
        />
      </a>
    </li>
  </ul>
</div>
`;
